test/integration/exchange/balancerV2ExchangeAdapter.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+    tradeModule = await deployer.modules.deployTradeModule(setup.controller.address);
+    await setup.controller.addModule(tradeModule.address);
+
+    await setup.integrationRegistry.addIntegration(
+      tradeModule.address,
+      balancerV2AdapterName,
+      balancerV2ExchangeAdapter.address
+    );
+
+    await initializeForkedTokens(deployer);
+  });
+
+  describe("#trade", function() {
+    let sourceToken: IERC20;
+    let destinationToken: IERC20;
+    let setToken: SetToken;
+    let issueQuantity: BigNumber;
+
+    context("when trading a component using BalancerV2 exchange adapter", async () => {
+      let mockPreIssuanceHook: ManagerIssuanceHookMock;
+
+      let subjectDestinationToken: Address;
+      let subjectSourceToken: Address;
+      let subjectSourceQuantity: BigNumber;
+      let subjectAdapterName: string;
+      let subjectSetToken: Address;
+      let subjectMinDestinationQuantity: BigNumber;
+      let subjectData: Bytes;
+      let subjectCaller: Account;
+
+      cacheBeforeEach(async () => {
+        tokens = getForkedTokens();
+
+        sourceToken = tokens.reth;
+        destinationToken = tokens.weth;
+
+        // Create Set token
+        setToken = await setup.createSetToken(
+          [sourceToken.address],
+          [ether(1)],
+          [setup.issuanceModule.address, tradeModule.address],
+          manager.address
+        );
+
+        const spender = await balancerV2ExchangeAdapter.getSpender();
+
+        await sourceToken.approve(spender, ether(100));
+        await destinationToken.approve(spender, ether(100));
+
+        tradeModule = tradeModule.connect(manager.wallet);
+        await tradeModule.initialize(setToken.address);
+
+        // Transfer from reth whale to manager
+        await sourceToken.transfer(manager.address, ether(20));
+
+        // Approve tokens to Controller and call issue
+        sourceToken = sourceToken.connect(manager.wallet);
+        await sourceToken.approve(setup.issuanceModule.address, ethers.constants.MaxUint256);
+
+        // Deploy mock issuance hook and initialize issuance module
+        setup.issuanceModule = setup.issuanceModule.connect(manager.wallet);
+        mockPreIssuanceHook = await deployer.mocks.deployManagerIssuanceHookMock();
+        await setup.issuanceModule.initialize(setToken.address, mockPreIssuanceHook.address);
+
+        issueQuantity = ether(5);
+        await setup.issuanceModule.issue(setToken.address, issueQuantity, owner.address);
+      });
+
+      beforeEach(async () => {
+        subjectSetToken = setToken.address;
+        subjectSourceToken = sourceToken.address;
+        subjectDestinationToken = destinationToken.address;
+        subjectSourceQuantity = ether(1);
+        subjectMinDestinationQuantity = ether(1); // assume almost 1:1
+        subjectAdapterName = balancerV2AdapterName;
+        subjectData = "0x1e19cf2d73a72ef1332c882f20534b6519be0276000200000000000000000112";
```

Comment:
Defining this constant before assigning it to the test fixture would be ideal. I couldn't tell you what these bytes represent. PoolID?

Reply:
yarp poolId - will do